File: 3차/1주차/김민정/포도주시식.py
from sys import stdin
input = stdin.readline

N = int(input())

wine = [0]

for _ in range(N):
    wine.append(int(input()))

dp = [0, wine[1]]

if N > 1:
    dp.append(wine[1] + wine[2])

for i in range(3, N + 1):
    dp.append(max(dp[i - 1], dp[i - 2] + wine[i], dp[i - 3] + wine[i - 1] + wine[i]))

print(dp[N])

File: 3차/1주차/고주희/포도주시식.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		
		int[][] wine = new int[N+1][3];
		
		for (int i = 1; i <= N; i++) {
			int n = Integer.parseInt(br.readLine());
			wine[i][0] = Math.max(Math.max(wine[i - 1][1], wine[i - 1][2]), wine[i - 1][0]);
			wine[i][1] = wine[i - 1][0] + n;
			wine[i][2] = wine[i - 1][1] + n;
		}
		
		System.out.println(Math.max(wine[N][0], Math.max(wine[N][1], wine[N][2])));
	}

}
